### PR TITLE
feat(api): Add analytics tracking for incidents (SEN-567)

### PR DESCRIPTION
**Pull request by wedamija**
This adds tracking to:
Incident creation
Incident status changes
Incident comments

Also adds `referrer=incident_activity_email` to the incident link in activity emails for clicktracking purposes.

### CODE REVIEW

File: tests/sentry/incidents/test_logic.py [hunk 1]
```diff
@@ -111,6 +130,18 @@ def run_test(
         assert activity.comment == comment
         assert activity.event_stats_snapshot is None
 
+        print self.record_event.calls
```
Comment (markstory) [inline tests/sentry/incidents/test_logic.py]: left over print.

**Review by markstory — COMMENTED** [review]
Changes look pretty good to me. Have these events been added to reload?

File: src/sentry/incidents/logic.py [hunk 2]
```diff
@@ -186,6 +204,16 @@ def create_incident_activity(
...
         countdown=10,
     )
+    if activity_type == IncidentActivityType.COMMENT:
+        analytics.record(
+            'incident.comment',
+            incident_id=incident.id,
+            organization_id=incident.organization_id,
+            incident_type=incident.type,
+            user_id=user.id,
+            comment=comment,
```
Comment (billyvg) [inline src/sentry/incidents/logic.py]: Do we want to record the actual comment?

Comment (wedamija) [reply src/sentry/incidents/logic.py]: Confirming with @saragilford

Comment (wedamija) [reply src/sentry/incidents/logic.py]: No need for the comment, removed

File: src/sentry/incidents/events.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+class BaseIncidentEvent(analytics.Event):
+    attributes = (
+        analytics.Attribute('incident_id'),
+        analytics.Attribute('organization_id'),
+        analytics.Attribute('incident_type'),
+    )
+
+
+class IncidentCreatedEvent(BaseIncidentEvent):
+    type = 'incident.created'
```
Comment (billyvg) [inline src/sentry/incidents/events.py]: FYI these need to be added here: https://github.com/getsentry/reload/blob/master/reload_app/events.py

Comment (wedamija) [reply src/sentry/incidents/events.py]: @billyvg Ahh will do. Does this need to happen before landing this pr?

Comment (billyvg) [reply src/sentry/incidents/events.py]: In general it should but since this is an internal feature atm it won't matter really

**Review by billyvg — APPROVED** [review]

**Pull request merged**